src/components/smartSearch/filters/TimeFrame.tsx

```diff
@@ -0,0 +1,120 @@
+import DateFnsUtils from '@date-io/date-fns';
+import { FormattedMessage as Msg } from 'react-intl';
+import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { MenuItem, Typography } from '@material-ui/core';
+import { useEffect, useState } from 'react';
+
+import { getNewDateWithOffset } from 'utils/dateUtils';
+import { getTimeFrame } from '../utils';
+import StyledDatePicker from '../inputs/StyledDatePicker';
+import StyledNumberInput from '../inputs/StyledNumberInput';
+import StyledSelect from '../inputs/StyledSelect';
+import { TIME_FRAME } from 'types/smartSearch';
+
+interface TimeFrameProps {
+    onChange: (range: {after?: string; before?: string}) => void;
+    filterConfig?: {after?: string; before?: string};
+}
+
+const TimeFrame = ({ onChange, filterConfig }: TimeFrameProps): JSX.Element => {
+    const [selected, setSelected] = useState<TIME_FRAME>(TIME_FRAME.EVER);
+    const today = new Date();
+    const [before, setBefore] = useState(today);
+    const [after, setAfter] = useState(getNewDateWithOffset(today, -30));
+    const [numDays, setNumDays] = useState(30);
+
+    useEffect(() => {
+        if (filterConfig) {
+            const timeFrame = getTimeFrame(filterConfig);
+            const { after, before } = filterConfig;
+            setSelected(timeFrame);
+            if (timeFrame === TIME_FRAME.LAST_FEW_DAYS && after) {
+                setNumDays(+after.substring(1, after.length - 1));
+                return;
+            }
+            else if (timeFrame === TIME_FRAME.BETWEEN && after && before) {
+                setAfter(new Date(after));
+                setBefore(new Date(before));
+            }
+            else if (timeFrame === TIME_FRAME.AFTER_DATE && after) {
+                setAfter(new Date(after));
+            }
+            else if (timeFrame === TIME_FRAME.BEFORE_DATE && before) {
+                setBefore(new Date(before));
+            }
+        }
+    }, []);// eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
That is all it's doing but the logic is really ugly when directly inside the setState

Reply:
`filterConfig?.before && (timeFrame ===  TIME_FRAME.BEFORE_DATE || timeFrame === TIME_FRAME.BETWEEN)) ? filterConfig.before : today`

Even if I destructure filterConfig, seems more convoluted to have the OR condition in the time frames rather than one long if else chain

Reply:
I would hide away the logic in a function, maybe augment `getTimeframe()` (or add another function) to give back not just the time frame enum value, but the default state for that configuration, e.g. something like this:

```tsx
const { timeframe, before, after, numDays } = getTimeframeConfig(filterConfig);
const [before, setBefore] = useState(before);
const [after, setAfter] = useState(after);
const [numDays, setNumDays] = useState(numDays)
```

Reply:
good idea